eng/QuarantinedTestRunsheetBuilder/QuarantinedTestRunsheetBuilder.targets

```diff
@@ -0,0 +1,195 @@
+      <_TestRunnerCommand Condition="'$(_TestRunnerAdditionalArguments)' != ''">$(_TestRunnerCommand) $(_TestRunnerAdditionalArguments)</_TestRunnerCommand>
+
+      <!--
+        Redirect std output of the runner.
+      -->
+      <_TestRunnerCommand Condition="'$(TestCaptureOutput)' != 'false'">$(_TestRunnerCommand) >> "%(TestToRun.ResultsStdOutPath)" 2>&amp;1</_TestRunnerCommand>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <_OutputFiles Include="%(TestToRun.ResultsStdOutPath)" />
+    </ItemGroup>
+
+    <MakeDir Directories="@(_OutputFiles->'%(RootDir)%(Directory)')"/>
+    <Delete Files="@(_OutputFiles)" />
+
+    <!--
+      Add command line to the log.
+      -->
+    <Exec Command="echo === COMMAND LINE === > %(TestToRun.ResultsStdOutPath)
+                   echo $(_TestRunnerCommand) >> %(TestToRun.ResultsStdOutPath)" />
+
+    <Message Text="Searching for quarantined tests: $(_TestAssembly) [$(_TestEnvironment)]" Importance="high"/>
+    <Exec Command='$(_TestRunnerCommand)'
+          LogStandardErrorAsError="false"
+          WorkingDirectory="$(_TargetDir)"
+          IgnoreExitCode="true"
+          Timeout="$(_TestTimeout)"
+          ContinueOnError="WarnAndContinue">
+      <Output TaskParameter="ExitCode" PropertyName="_TestErrorCode" />
+    </Exec>
+
+    <!--
+      Report test status.
+      -->
+    <Message Text="Search complete: $(_TestAssembly) [$(_TestEnvironment)]" Condition="'$(_TestErrorCode)' == '0'" Importance="high" />
+
+    <PropertyGroup>
+      <_ResultsFileToDisplay>%(TestToRun.ResultsStdOutPath)</_ResultsFileToDisplay>
+    </PropertyGroup>
+
+    <!--
+      Ideally we would set ContinueOnError="ErrorAndContinue" so that when a test fails in multi-targeted test project
+      we'll still run tests for all target frameworks. ErrorAndContinue doesn't work well on Linux though: https://github.com/Microsoft/msbuild/issues/3961.
+    -->
+    <Error Text="Search failed: $(_ResultsFileToDisplay) [$(_TestEnvironment)]" Condition="'$(_TestErrorCode)' != '0'" File="VSTest" />
+
+    <!--
+      Read the test result and see if the string "No test matches the given testcase filter" is present.
+      This indicates that there are no quarantined tests to run.
+      -->
+    <ReadLinesFromFile File="$(_ResultsFileToDisplay)">
+      <Output TaskParameter="Lines" ItemName="FileLines" />
+    </ReadLinesFromFile>
+
+    <ItemGroup>
+      <_NoTestMatchesLines Include="@(FileLines)" Condition=" $([MSBuild]::ValueOrDefault(`%(FileLines.Identity)`, '').Contains(`No test matches the given testcase filter`)) " />
+    </ItemGroup>
+
+    <PropertyGroup>
+      <_HasQuarantinedTests>true</_HasQuarantinedTests>
+      <_HasQuarantinedTests Condition=" '@(_NoTestMatchesLines)' != '' ">false</_HasQuarantinedTests>
+    </PropertyGroup>
+
+    <Message Text="💡 Quarantined tests found in $(_TestAssembly) [$(_TestEnvironment)]" Condition=" '$(_HasQuarantinedTests)' == 'true' " Importance="high" />
+
+    <!--
```

Comment:
I would suggest splitting this target into two:
1. `RunTests` which would contain the second half which generates the runsheets. And that can be conditioned on ` ('$(RunOnGithubActionsWindows)' == 'true' or '$(RunOnGithubActionsLinux)' == 'true') and  '$(_HasQuarantinedTests)' == 'true'`.

2. And `RunTests` would depend on a target `GetQuarantinedTests` which would contain the bits to get the list of quarantined tests.

Reply:
I've been considering this, but this is more in a "nice to have" bucket. It would require additional work and validation, and I don't have time for it right now.